src/core/image.js

```diff
@@ -641,6 +643,35 @@ var PDFImage = (function PDFImageClosure() {
         this.undoPreblend(imgData.data, drawWidth, actualHeight);
       }
 
+      if (transferFnsIR) {
+        var numTransferFnsIR = transferFnsIR.length;
+        var numChannels = alpha01 === 0 ? 3 : 4;
+
+        if (numTransferFnsIR < numChannels) {
+          warn('Too few transfer functions.');
+          return imgData;
+        }
+        var transferFnIR, transferFn, j, jj;
+        var transferMap = new Uint8Array(256), tmp = new Float32Array(1);
```

Comment:
> (I'm thinking not support transfer function for other CSs yet)

Based on the additional PDF files provided in https://github.com/mozilla/pdf.js/issues/6931#issuecomment-178693440, we probably need to implement this more generally.

I'll readily admit that the patch, as is stands needs a fair bit of improvement :-)
However, I'm having a bit of trouble envisioning how you'd prefer this to be implemented. E.g. how to best divide the code between `evaluator.js`/`image.js`. Some additional pointer would be much appreciated here!